torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -461,6 +461,14 @@ class AsyncAllreduceWork : public ProcessGroupGloo::AsyncWork {
 
   void run() override {
     allreduce(inputs);
+
+    // Only the first output in the tensor list contains the results.
+    // See https://github.com/facebookincubator/gloo/issues/152.
+    // The contents is the same for every entry in the tensor list, so
+    // we can use the first entry as the source of the copy below.
+    for (size_t i = 1; i < inputs.size(); i++) {
```

Comment:
Hmmmm, are we sure this is the right fix? This is basically 7 copies for 8 GPUs?